llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -36,7 +37,7 @@ def __init__(
         # Broker machinery
         self._queues: Dict[str, asyncio.Queue] = {}
         self._tasks: Set[asyncio.Task] = set()
-        self._broker_log: List[Event] = []
+        self._broker_log: List[Checkpoint] = []
```

Comment:
being a private field I think we can change its name, feel free to call it something like `_checkpoints`

Reply:
Also I was wondering, should we make this a list of lists (each item being checkpoints from a single run) or a dictionary (with keys being an "identifier" of the run, or a timestamp)?

Reply:
Right, will change `_broker_log` to `_checkpoints` then instead. I do agree that we should opt for some structure to be able to distinguish between runs. Will add that in.

Reply:
Changed this to a `dict[str, list[Checkpoint]]` where keys are `run_id`'s